src/parsers/multitest.js

```diff
@@ -1,29 +1,26 @@
-const PROTRACTOR_SHARDED = /------------------------------------/g
-const SPECFILE_REG = /.+Specs:\s(.*\.js)/g
+const PROTRACTOR_MULTITEST = /------------------------------------/g
+const SPECFILE_REG = /Specs:\s(.*\.js)/g
 
 export default {
-  name: 'ShardedParser',
+  name: 'MultiTestParser',
   test (output) {
-    return PROTRACTOR_SHARDED.test(output) && SPECFILE_REG.test(output)
+    return PROTRACTOR_MULTITEST.test(output) && SPECFILE_REG.test(output)
   },
 
   parse (failedSpecs, output) {
     let match = null
-    let testsOutput = output.split('------------------------------------')
-    testsOutput.shift()
-    let RESULT_REG = /\d+\sspec|assertions?,\s(\d+)\sfailures?/g
+    let testsOutput = output.split('------------------------------------').slice(1)
+    let RESULT_REG = /,\s0 failures/g
     testsOutput.forEach(function (test) {
       let specfile
       let result = 'failed'
       // retrieve specfile from run;
       while (match = SPECFILE_REG.exec(test)) { // eslint-disable-line no-cond-assign
         specfile = match[1]
       }
-      // check for string 'X specs, X failures' and verify that failures === 0;
+      // check for string '0 failures' and then marks the test as passed
       while (match = RESULT_REG.exec(test)) { // eslint-disable-line no-cond-assign
```

Comment:
The difference lies in the following situation

```
Intro
----
Specs: filename.js
1 failure
----
No filename given
1 failure
----
Specs: anotherTest.js
0 failures
Overall failed with exit code 153
```

Protractor-flake will now re-run filename.js and skip the 'No filename given' test.
But this is a situation that isn't fixed with option A either. I was just brainstorming if this could be handled here somehow.
Do you have any great ideas? Else I'll just implement option B.